### PR TITLE
enh(topCounter): Redirect top counter links to the resource status page

### PR DESCRIPTION
**Pull request by bdauria**
## Description

This replaces the resource links in the top counter for both hosts and services to the resources status page instead of the old monitoring ones.

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)

## Target serie

- [x] 20.10.x (master)

<h2> How this pull request can be tested ? </h2>
- check that all the links for host and service menus in the top counter redirects to the resources status page with the correct statuses.

### CODE REVIEW

**Comment by Thebarda** [comment]
LGTM with latest changes ✔️ 
topcounter

**Review by Thebarda — DISMISSED** [review]

File: www/front_src/src/components/header/getResourcesUrl.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+});
+const upCriterias = getStatusCriterias({ id: 'UP', name: 'Up' });
+const pendingCriterias = getStatusCriterias({ id: 'PENDING' });
+const criticalCriterias = getStatusCriterias({
+  id: 'CRITICAL',
+  name: 'Critical',
+});
+const warningCriterias = getStatusCriterias({ id: 'WARNING', name: 'Warning' });
+const unknownCriterias = getStatusCriterias({ id: 'UNKNOWN', name: 'Unknown' });
+const okCriterias = getStatusCriterias({ id: 'OK', name: 'Ok' });
```
Comment (jdelpierre) [inline www/front_src/src/components/header/getResourcesUrl.ts:25]: All this status information is common in Centreon. Shouldn't we put them somewhere more general so as not to define them several times?

File: www/front_src/src/components/header/getResourcesUrl.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  return { statuses: [status] };
+};
+
+const downCriterias = getStatusCriterias({ id: 'DOWN', name: 'Down' });
+const unreachableCriterias = getStatusCriterias({
+  id: 'UNREACHABLE',
+  name: 'Unreachable',
+});
+const upCriterias = getStatusCriterias({ id: 'UP', name: 'Up' });
+const pendingCriterias = getStatusCriterias({ id: 'PENDING' });
```
Comment (jdelpierre) [inline www/front_src/src/components/header/getResourcesUrl.ts:18]: no "name" in this object?

**Review by jdelpierre — APPROVED** [review]

File: www/front_src/src/components/header/serviceStatusMenu/index.js [hunk 3]
```diff
@@ -148,7 +156,7 @@ class ServiceStatusMenu extends Component {
           </IconHeader>
           <Link
             className={classnames(styles.link, styles['wrap-middle-icon'])}
-            to="/main.php?p=20201&o=svc_unhandled&statusFilter=critical&search="
+            to={getServiceResourcesUrl(criticalCriterias)}
```
Comment (kduret) [inline www/front_src/src/components/header/serviceStatusMenu/index.js:159]: `unhandled problems` filter should also be set

File: www/front_src/src/components/header/serviceStatusMenu/index.js [hunk 4]
```diff
@@ -164,7 +172,7 @@ class ServiceStatusMenu extends Component {
           </Link>
           <Link
             className={classnames(styles.link, styles['wrap-middle-icon'])}
-            to="/main.php?p=20201&o=svc_unhandled&statusFilter=warning&search="
+            to={getServiceResourcesUrl(warningCriterias)}
```
Comment (kduret) [inline www/front_src/src/components/header/serviceStatusMenu/index.js:175]: `unhandled problems` filter should also be set

File: www/front_src/src/components/header/serviceStatusMenu/index.js [hunk 5]
```diff
@@ -180,7 +188,7 @@ class ServiceStatusMenu extends Component {
           </Link>
           <Link
             className={classnames(styles.link, styles['wrap-middle-icon'])}
-            to="/main.php?p=20201&o=svc_unhandled&statusFilter=unknown&search="
+            to={getServiceResourcesUrl(unknownCriterias)}
```
Comment (kduret) [inline www/front_src/src/components/header/serviceStatusMenu/index.js:191]: `unhandled problems` filter should also be set

File: www/front_src/src/components/header/serviceStatusMenu/index.js [hunk 6]
```diff
@@ -196,7 +204,7 @@ class ServiceStatusMenu extends Component {
           </Link>
           <Link
             className={classnames(styles.link, styles['wrap-middle-icon'])}
-            to="/main.php?p=20201&o=svc&statusFilter=ok&search="
+            to={getServiceResourcesUrl(okCriterias)}
```
Comment (kduret) [inline www/front_src/src/components/header/serviceStatusMenu/index.js:207]: `unhandled problems` filter should also be set

File: www/front_src/src/components/header/serviceStatusMenu/index.js [hunk 7]
```diff
@@ -255,7 +263,7 @@ class ServiceStatusMenu extends Component {
                 />
               </Link>
               <Link
-                to="/main.php?p=20201&o=svc_unhandled&statusFilter=unknown&search="
+                to={getServiceResourcesUrl(unknownCriterias)}
```
Comment (kduret) [inline www/front_src/src/components/header/serviceStatusMenu/index.js:266]: `unhandled problems` filter should also be set

File: www/front_src/src/components/header/serviceStatusMenu/index.js [hunk 8]
```diff
@@ -242,7 +250,7 @@ class ServiceStatusMenu extends Component {
                 />
               </Link>
               <Link
-                to="/main.php?p=20201&o=svc_unhandled&statusFilter=warning&search="
+                to={getServiceResourcesUrl(warningCriterias)}
```
Comment (kduret) [inline www/front_src/src/components/header/serviceStatusMenu/index.js:253]: `unhandled problems` filter should also be set

File: www/front_src/src/components/header/serviceStatusMenu/index.js [hunk 9]
```diff
@@ -229,7 +237,7 @@ class ServiceStatusMenu extends Component {
                 />
               </Link>
               <Link
-                to="/main.php?p=20201&o=svc_unhandled&statusFilter=critical&search="
+                to={getServiceResourcesUrl(criticalCriterias)}
```
Comment (kduret) [inline www/front_src/src/components/header/serviceStatusMenu/index.js:240]: `unhandled problems` filter should also be set

File: www/front_src/src/components/header/getResourcesUrl.ts (hunk 2 above)
Comment (bdauria) [reply www/front_src/src/components/header/getResourcesUrl.ts:18]: Actually no name is required in any object, i'll remove them.

File: www/front_src/src/components/header/getResourcesUrl.ts (hunk 1 above)
Comment (bdauria) [reply www/front_src/src/components/header/getResourcesUrl.ts:25]: I tend to prefer to wait for another use case to generalize any implementation

File: www/front_src/src/components/header/serviceStatusMenu/index.test.tsx [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+    store.dispatch(setRefreshIntervals({ AjaxTimeReloadMonitoring: 10 }));
+
+    await waitFor(() => {
+      expect(mockedAxios.get).toHaveBeenCalled();
+    });
+
+    fireEvent.click(getByText('4'));
+    expect(window.location.href).toBe(
+      'http://localhost/monitoring/resources?filter={%22criterias%22:{%22resourceTypes%22:[{%22id%22:%22service%22}],%22statuses%22:[{%22id%22:%22CRITICAL%22}],%22states%22:[{%22id%22:%22unhanlded_problems%22}]}}&fromTopCounter=true',
+    );
```
Comment (jdelpierre) [inline www/front_src/src/components/header/serviceStatusMenu/index.test.tsx:61]:
```suggestion
      'http://localhost/monitoring/resources?filter={"criterias":{"resourceTypes":[{"id":"service"}],"statuses":[{"id":"CRITICAL"}],"states":[{"id":unhanlded_problems"}]}}&fromTopCounter=true',
    );
```
It is not possible to delete the encoded spaces?

Comment (bdauria) [reply www/front_src/src/components/header/serviceStatusMenu/index.test.tsx:61]: actually it's not, since that links are encoded out of the box

**Review by kduret — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

**Comment by Thebarda** [comment]
LGTM with latest changes ✔️

**Review by Thebarda — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

**Pull request merged**